lib/led-matrix/src/LEDMatrix.cpp

```diff
@@ -0,0 +1,154 @@
+/* LED Matrix Source File */
+
+// naming with m_
+#include "LEDMatrix.h"
+
+#include "mbed.h"
+
+// init thread ptr to nullptr and initialize the requestEndFlashing semaphore to 0 and a max of 1
+LEDMatrix::LEDMatrix(PinName R, PinName G, PinName B)
+    : m_RChannel(R), m_GChannel(G), m_BChannel(B), lightsThread(nullptr), reqEndFlash(0, 1) {}
+
+void LEDMatrix::terminateFlashing() {
+  reqEndFlash.release();  // ask flashing thread to end - signal semaphore
+  lightsThread->join();   // wait for the thread to finish. we've asked it to finish so should be quick wait
+  lightsThread->terminate();
+  delete lightsThread;
+  lightsThread = nullptr;
+}
+void LEDMatrix::flashing() {
+  while (!reqEndFlash.try_acquire()) {  // while we havent been asked to terminate this thread
+    printf("Flashing thread\r\n");
+    // do flashing with flashing_color which is a member var
+
+    while (true) {
```

Comment:
Outer condition never gets checked since theres a while true here 🤔

Reply:
Right, it would only get checked the one time... I guess I don't need the `while (true)`